pypuf/simulation.py

```diff
@@ -106,3 +116,37 @@ def ltf_eval(self, inputs):
                 for l in range(self.k)
             ])
         )
+
+
+class NoisyLTFArray(LTFArray):
+    """
+    Class that simulates k LTFs with n bits and a constant term each 
+    with noise effect and constant bias (=scale) added.
+    """
+    def __init__(self, weight_array, transform, combiner, weight_scale,
+                 noisiness=0.1, noise_seed=None, bias=False):
+        """
+        Initializes LTF array with bias and noise effect based on inputs.
+        The noise effect is a normal distributed random variable
+        with mu=0, sigma= scale of overall weights * noisiness. 
+        It can be seeded optionally by a value out of [0, 2^32 -1].
+        """
+        if (noise_seed != None):
+            random.seed(noise_seed)
+        super().__init__(weight_array, transform, combiner, bias)
+        self.sigma_noise = sqrt(self.n) * weight_scale * noisiness
```

Comment:
What is the purpose of using two user-chosen parameters `weight_scale` and `noisiness`?